inflect.py

```diff
@@ -2286,8 +2286,19 @@ def inflect(self, text: str) -> str:
     def postprocess(self, orig: str, inflected) -> str:
         inflected = str(inflected)
         if "|" in inflected:
-            inflected = inflected.split("|")[self.classical_dict["all"]]
-        result = inflected.split(" ")
+            word_options = inflected.split("|")
+            # When two parts of a noun need to be pluralized
+            if len(word_options[0].split(" ")) == len(word_options[1].split(" ")):
+                result = inflected.split("|")[self.classical_dict["all"]].split(" ")
+            # When only the last part of the noun needs to be pluralized
+            else:
+                result = inflected.split(" ")
+                for index, word in enumerate(result):
+                    if "|" in word:
+                        result[index] = word.split("|")[self.classical_dict["all"]]
```

Comment:
I was originally thinking that this approach could be replaced with a comprehension, something like this:

```suggestion
                result = [
                    word.split("|")[self.classical_dict["all"]] if "|" in word else word
                    for word in inflected.split(" ")
                ]
```

But I'm not sure that reads any better. What do you think?

Reply:
It's definitely more concise. Hadn't seen exactly that type of comprehension before, so had to read it a couple of times – but I think it works fine here.

Reply:
In that case, let's just go with the original approach.